### DISCUSSION TITLE
Imgur upload failed, check dev console

### DISCUSSION THREAD

**Discussion by LynnXie00** [gavvvr/obsidian-imgur-plugin#37] [Getting help]
I'm using the authenticated method to upload images. But all of a sudden it stops to upload any image, with a warning saying 

`<!--⚠️Imgur upload failed, check dev console-->`

I love this plugin very much so far. Wanting to know what I can do to fix it...

**Comment by gavvvr** [comment]
Hi @LynnXie00
Did the authenticated upload work fine before? 
2 days ago I stared using authenticated upload and I also face this issue in ~50% of uploads. But I do not even know if it was always like that, or it's just a coincidence that it started failing recently? Previously I was always using anonymous uploads.

Anyway, I do not have any idea why it fails. When the request fails, the console says it was blocked by CORS... From my point of view it should either 100% work or should be 100% CORS blocked. But in practice the same request can fail and then be successful if you try again 🤯 That's weird

**Reply by LynnXie00** [reply]
Thank you for your reply! It seems working now... Not sure why it sometimes does not work... It happens last time when I was trying to copy my notes somewhere else to Obsidian, maybe because I uploaded too many pictures at once so Imgur thinks I'm suspicious... Does their policy has some sort of detection for these things?

**Reply by gavvvr** [reply]
There are some [limits](https://apidocs.imgur.com) (see __Rate Limits__). But I am not sure you could reach them...

**Reply by LynnXie00** [reply]
> There are some [limits](https://apidocs.imgur.com) (see **Rate Limits**). But I am not sure you could reach them...

I see, 1,250 uploads per day, that looks like a lot of images...I don't think I have uploaded that many pictures, but since my setting is to upload by default, the limit could be reached without me realizing it... Maybe it should be put in the documentation of the plugin as a reminder that there is a limit... or it could also be a good feature to set image uploaded only for notes in certain folders. I think a lot of people want to upload images because they want to publish the notes.

**Reply by LynnXie00** [reply]
> There are some [limits](https://apidocs.imgur.com) (see **Rate Limits**). But I am not sure you could reach them...

Oh it fails to work again... now I'm sure it's not over 1250 images today....

**Reply by gavvvr** [reply]
> Maybe it should be put in the documentation of the plugin as a reminder that there is a limit...

Already [mentioned in README](https://github.com/gavvvr/obsidian-imgur-plugin/blob/2.2.1/README.md#known-issues)

> it could also be a good feature to set image uploaded only for notes in certain folders

This is an interesting idea. Nobody has requested this so far. I will make a note
